src/fcc_bridge_node_ros.cpp

```diff
@@ -0,0 +1,225 @@
+    if (!msg.active) {
+        RCLCPP_ERROR(this->get_logger(),
+                     "Mission control node is inactive! Triggering RTH...");
+        this->trigger_rth();
+    }
+
+    // Set the cached heartbeat message to the current one
+    this->last_mission_control_heartbeat = msg;
+
+    // Verify that the received heartbeat is not too old
+    this->check_last_mission_control_heatbeat();
+
+    RCLCPP_INFO(this->get_logger(), "Mission control is alive and ok");
+}
+
+void FCCBridgeNode::fcc_telemetry_timer_5hz_cb() {
+    RCLCPP_DEBUG(this->get_logger(),
+                 "5Hz telemetry timer callback was triggered");
+    this->check_last_mission_control_heatbeat();
+    switch (this->internal_state) {
+        case INTERNAL_STATE::STARTING_UP:
+        case INTERNAL_STATE::ROS_SET_UP:
+        case INTERNAL_STATE::ERROR:
+            RCLCPP_WARN(this->get_logger(),
+                        "5Hz Telemetry callback function was called in an "
+                        "invalid state");
+            return;
+        default:
+            break;
+    }
+
+    // Update flight state
+    this->get_flight_state();
+    interfaces::msg::FlightState flight_state_msg;
+    flight_state_msg.time_stamp = this->now();
+    flight_state_msg.sender_id = this->get_name();
+    flight_state_msg.flight_mode = FCCBridgeNode::flight_mode_mavsdk_to_ros(
+        this->last_fcc_flight_state.value());
+    this->flight_state_publisher->publish(flight_state_msg);
+    RCLCPP_DEBUG(this->get_logger(), "Published current Flight State");
+
+    // Update battery state
+    this->get_battery_state();
+    interfaces::msg::BatteryState battery_state_msg;
+    battery_state_msg.time_stamp = this->now();
+    battery_state_msg.sender_id = this->get_name();
+    battery_state_msg.id = this->last_fcc_battery_state->id;
+    battery_state_msg.temperature_degc =
+        this->last_fcc_battery_state->temperature_degc;
+    battery_state_msg.voltage_v = this->last_fcc_battery_state->voltage_v;
+    battery_state_msg.current_battery_a =
+        this->last_fcc_battery_state->current_battery_a;
+    battery_state_msg.capacity_consumed_ah =
+        this->last_fcc_battery_state->capacity_consumed_ah;
+    battery_state_msg.remaining_percent =
+        this->last_fcc_battery_state->remaining_percent;
+    this->battery_state_publisher->publish(battery_state_msg);
+    RCLCPP_DEBUG(this->get_logger(), "Published current battery state");
+
+    // Update RC state
+    this->get_rc_state();
+    interfaces::msg::RCState rc_state_msg;
+    rc_state_msg.time_stamp = this->now();
+    rc_state_msg.sender_id = this->get_name();
+    rc_state_msg.was_available_once =
+        this->last_fcc_rc_state->was_available_once;
+    rc_state_msg.is_available = this->last_fcc_rc_state->is_available;
+    rc_state_msg.signal_strength_percent =
+        this->last_fcc_rc_state->signal_strength_percent;
+    this->rc_state_publisher->publish(rc_state_msg);

```

Comment:
Did you select the wrong line. I suspect you wanted to reference the same as in https://github.com/THI-Drone/fcc_bridge_package/pull/1#discussion_r1587404471
The default case is empty to fall through instead of putting all the logic into the switch case I put it after it.
Without the default case there is an warning->error.

Reply:
Understood. But why not put the code inti the switch case (of course jn the case for the correct cases with fallthrough to onky write it once).

Reply:
I didn´t want to put all that code into the scope of the switch case. So I opted to move it into the function scope and hint with the break, that execution will continue in those cases. The compiler should optimize that away.
At least I think it makes the code more readable because the switch case is of no concern when it comes to the latter code.